lib/views/item-scene.js

```diff
@@ -22,6 +22,49 @@ class ItemScene extends React.Component {
   constructor(props) {
     super(props);
     debug('created');
+
+    let distance = this.props.item.distance;
+    if (typeof distance !== 'number') {
+      distance = null;
+    }
+
+    this.state = {
+      distance,
+    };
+  }
+
+  componentDidMount() {
+    if (this.state.distance !== null ||
+      typeof this.props.item.latitude !== 'number' ||
+      typeof this.props.item.longitude !== 'number') {
+      return;
+    }
+
+    navigator.geolocation.getCurrentPosition((pos) => {
+        const distance = this.computeDistance(pos.coords);
+        this.setState({ distance });
+      },
+      console.log
+    );
+  }
+
+  computeDistance({ latitude, longitude }) {
```

Comment:
Should we add a timer to call this `computeDistance` periodically meanwhile you are in the POI? Every  1 minute or something like that?